### PR TITLE
Apikey

### PR DESCRIPTION
**Pull request by ngeissel**
Greetings,

Your work was amazing on this API. I am using your package in another project.

Unfortunately, FMP has become limited and now requires an API key :
- handle api key
- update readme

TODO : 
- control if is ok
- update tests (i did not succeed to adapt them 👎 )
- up version

You can find in this pull request a proposal to handle this one without too much changes. 
I am not an expert so please, feel free to do your feedback. 
Tell me if this is enough to publish a new version of npm.

Many thanksRegards,

Nicolas

### CODE REVIEW

**Comment by ngeissel** [comment]
Hello, i did a last change on index using mutualfund (you never used it neither)

File: index.js [hunk 1]
```diff
@@ -4,18 +4,24 @@ const commodities = require('./lib/commodities');
...
+FinancialModelingPrep.prototype.stock = stock;
+FinancialModelingPrep.prototype.market = market;
+FinancialModelingPrep.prototype.forex = forex;
+FinancialModelingPrep.prototype.crypto = crypto;
+FinancialModelingPrep.prototype.search = search;
+FinancialModelingPrep.prototype.commodities = commodities;
+FinancialModelingPrep.prototype.etf = etf;
+FinancialModelingPrep.prototype.mutualfund = mutualfund;
+
+module.exports = FinancialModelingPrep;
```
Comment (patelneel55) [inline index.js:27]:
```suggestion
const FinancialModelingPrep = function FinancialModelingPrep(apikey) {
  auth.key = apikey;
};

FinancialModelingPrep.prototype.stock = stock;
FinancialModelingPrep.prototype.market = market;
FinancialModelingPrep.prototype.forex = forex;
FinancialModelingPrep.prototype.crypto = crypto;
FinancialModelingPrep.prototype.search = search;
FinancialModelingPrep.prototype.commodities = commodities;
FinancialModelingPrep.prototype.etf = etf;
FinancialModelingPrep.prototype.mutualfund = mutualfund;

module.exports = (apikey) => {
  auth.key = apikey;
  return {	
      stock: stock,
      market: market,
      forex: forex,
      crypto: crypto,	
      search: search,
      commodities: commodities,
      etf: etf,
      mutualfund: mutualfund
  }
}
```

**Review by patelneel55 — CHANGES_REQUESTED** [review]
Thanks for the changes @ngeissel!

Just a small nitpick, it would be great if package users could provide their API key directly into the `require` header.
Something like
```
const fmp = require('financialmodelingprep')(apikey) 
```

No need to worry about the tests, thats WIP from my end.

File: index.js (hunk 1 above)
Comment (patelneel55) [inline index.js]:
```suggestion
module.exports = (apikey) => {
  auth.key = apikey;
  return {	
      stock: stock,
      market: market,
      forex: forex,
      crypto: crypto,	
      search: search,
      commodities: commodities,
      etf: etf,
      mutualfund: mutualfund
  }
}
```

**Review by patelneel55 — CHANGES_REQUESTED** [review]
Thanks for the changes @ngeissel!

Just a small nitpick, it would be great if package users could provide their API key directly into the `require` header.
Something like
```
const fmp = require('financialmodelingprep')(apikey) 
```

No need to worry about the tests, thats WIP from my end.

**Pull request merged**